### ISSUE TITLE
Alc.exe 4.0.2.23975 crashes on StackOverflowException

### ISSUE THREAD

**Issue by navdotnetreqs** [microsoft/AL#5238]
Good day,

I'm running docker image bcinsider.azurecr.io/bcsandbox-master:fi-ltsc2019 pulled today.

I've unzipped the included al-compiler (4.0.2.23975) and am attempting to compile our extension. 

As long as there were problems due to breaking changes, the compiler run ok. But now that everything should be fixed, the compiler crashes on a StackOverFlowException. I get no further details of the error..

![al_crash](https://user-images.githubusercontent.com/35798628/63777345-9f7ab680-c8eb-11e9-8327-e69e1d8f7e7d.png)

**Comment by atoader** [comment]
@navdotnetreqs let's try to narrow this down. Can you disable code analysis and see if it still reproes? Is there any additional information in the exception? You should see a stracktrace.

**Comment by navdotnetreqs** [comment]
Hi again,

I'm afraid this seems to be hard to reproduce. I retried the compilation with the same result a few times, but then the whole container crashed. I had to restart it, and now the extension compiles fine. This might have been related to old .app-files (from different versions of this extension and dependency extensions) that existed in the packagecache, which was now cleared upon restart.

![container_crash](https://user-images.githubusercontent.com/35798628/63785112-7b71a200-c8f8-11e9-8673-be565ddc9991.png)
